libraries/bezier-rs/src/subpath/lookup.rs

```diff
@@ -10,6 +12,65 @@ impl Subpath {
 		self.iter().fold(0., |accumulator, bezier| accumulator + bezier.length(num_subdivisions))
 	}
 
+	fn global_euclidean_to_local_euclidean(&self, global_t: f64) -> (usize, f64) {
+		let lengths = self.iter().map(|bezier| bezier.length(None)).collect::<Vec<f64>>();
+		let total_length: f64 = lengths.iter().sum();
+
+		let mut accumulator = 0.;
+		for (index, length) in lengths.iter().enumerate() {
+			let length_ratio = length / total_length;
+			if accumulator <= global_t && global_t <= accumulator + length_ratio {
+				return (index, (global_t - accumulator) / length_ratio);
+			}
+			accumulator += length_ratio;
+		}
+		(0, 0.)
+	}
+
+	/// Convert a [SubpathTValue] to a parametric `(segment_index, t)` tuple.
```

Comment:
Can you add to this comment to assert that the returned value will not refer to a non-existent segment index?